homeassistant/components/thethingsnetwork/sensor.py

```diff
@@ -80,10 +80,7 @@ def name(self):
     def state(self):
         """Return the state of the entity."""
         if self._ttn_data_storage.data is not None:
-            try:
-                return round(self._state[self._value], 1)
-            except (KeyError, TypeError):
-                return None
+            return self._state[self._value]
```

Comment:
The `try...except` block was added to fix #23097. It should not be removed otherwise we are back to the side effects.

Reply:
You're right. The `try...except` block must remain, but the except can be reduced to `except KeyError`.
I've fixed this with a new commit.